src/io/cloud.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * Cloud IO Device responsible for sending and receiving messages from
+ * cloud provider (mananging the cloud server connection) and interacting
+ * with cloud variables in the current project.
+ */
+
+const Variable = require('../engine/variable');
+const log = require('../util/log');
+
+class Cloud {
+    constructor () {
+        /**
+         * Reference to the cloud data provider, responsible for mananging
+         * the web socket connection to the cloud data server.
+         * @type{!object}
+         */
+        this.provider = null;
+
+        /**
+         * Reference to the stage target which owns the cloud variables
+         * in the project.
+         * @type{!Target}
+         */
+        this.stage = null;
+    }
+
+    /**
+     * Set a reference to the cloud data provider.
+     * @param {object} provider The cloud data provider
+     */
+    setProvider (provider) {
+        this.provider = provider;
+    }
+
+    /**
+     * Set a reference to the stage target which owns the
+     * cloud variables in the project.
+     * @param {Target} stage The stage target
+     */
+    setStage (stage) {
+        this.stage = stage;
+    }
+
+    /**
+     * Handle incoming data to this io device.
+     * @param {object} data The data to process
```

Comment:
Would you please document the shape data is expected to be?  I gave it a try just to demonstrate the format for objects. I don't know if this syntax is actually valid though — you may need to use `typedef` to document and refer to the shape of `varUpdate` (but then you could reuse it below)
```suggestion
     * @param {object} data The data to process
     * @param {object} data.varUpdate The update object for a variable
     * @param {string} data.varUpdate.name The name of the variable to update
     * @param {number} data.varUpdate.value The value of the variable to update
```